WalletWasabi/Stores/IndexStore.cs

```diff
@@ -44,33 +43,27 @@ public IndexStore(string workFolderPath, Network network, SmartHeaderChain hashC
 
 	public event EventHandler<FilterModel>? NewFilter;
 
-	private AbandonedTasks AbandonedTasks { get; } = new AbandonedTasks();
+	private AbandonedTasks AbandonedTasks { get; } = new();
 
 	private string WorkFolderPath { get; }
 	private Network Network { get; }
 	private DigestableSafeIoManager MatureIndexFileManager { get; }
 	private DigestableSafeIoManager ImmatureIndexFileManager { get; }
 
-	/// <summary>
-	/// Lock for accessing MatureIndex file. This should be lock #2.
-	/// </summary>
-	private AsyncLock MatureIndexAsyncLock { get; } = new AsyncLock();
+	/// <summary>Lock for modifying <see cref="ImmatureFilters"/>. This should be lock #1.</summary>
+	private AsyncLock IndexLock { get; } = new();
 
-	/// <summary>
-	/// Lock for accessing ImmatureIndex file. This should be lock #3.
-	/// </summary>
-	private AsyncLock ImmatureIndexAsyncLock { get; } = new AsyncLock();
+	/// <summary>Lock for accessing <see cref="MatureIndexFileManager"/>. This should be lock #2.</summary>
+	private AsyncLock MatureIndexAsyncLock { get; } = new();
+
+	/// <summary>Lock for accessing <see cref="ImmatureIndexFileManager"/>. This should be lock #3.</summary>
+	private AsyncLock ImmatureIndexAsyncLock { get; } = new();
 
 	public SmartHeaderChain SmartHeaderChain { get; }
 
 	private FilterModel StartingFilter { get; }
 	private uint StartingHeight => StartingFilter.Header.Height;
-	private List<FilterModel> ImmatureFilters { get; } = new List<FilterModel>(150);
-
-	/// <summary>
-	/// Lock for modifying SmartHeaderChain or ImmatureFilters. This should be lock #1.
-	/// </summary>
-	private AsyncLock IndexLock { get; } = new AsyncLock();
```

Comment:
(Just moved up so that the locks are sorted by the order in which they should be used.)